avm/utilities/pipelines/platform/Set-AvmGithubIssueForWorkflow.ps1

```diff
@@ -0,0 +1,128 @@
+<#
+.SYNOPSIS
+Check for failing pipelines and create issues for those, that are failing.
+
+.DESCRIPTION
+If a pipeline fails, a new issue will be created, with a link to the failed pipeline. If the issue is already existing, a comment will be added, if a new run failed (with the link for the new failed run). If a pipeline run succeeds and an issue is open for the failed run, it will be closed (and a link to the successful run is added to the issue).
+
+.PARAMETER Repo
+Mandatory. The name of the respository to scan. Needs to have the structure "<owner>/<repositioryName>"
+
+.PARAMETER LimitNumberOfRuns
+Optional. Number of recent runs to scan for failed runs. Default is 100.
+
+.PARAMETER LimitInDays
+Optional. Only runs in the past selected days will be analyzed. Default is 2.
+
+.PARAMETER IgnoreWorkflows
+Optional. List of workflow names that should be ignored (even if they fail, no ticket will be created). Default is an empty array.
+
+.EXAMPLE
+Set-AvmGithubIssueForWorkflow -Repo 'owner/repo01' -LimitNumberOfRuns 100 -LimitInDays 2 -IgnoreWorkflows @('Pipeline 01')
+
+Check the last 100 workflow runs in the repository 'owner/repo01' that happened in the last 2 days. If the workflow name is 'Pipeline 01', then ignore the workflow run.
+
+.NOTES
+The function requires GitHub CLI to be installed.
+#>
+function Set-AvmGithubIssueForWorkflow {
+  [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string] $Repo,
+
+    [Parameter(Mandatory = $false)]
+    [int] $LimitNumberOfRuns = 100,
+
+    [Parameter(Mandatory = $false)]
+    [int] $LimitInDays = 2,
+
+    [Parameter(Mandatory = $false)]
+    [String[]] $IgnoreWorkflows = @()
+  )
+
+  $issues = gh issue list --state open --label 'Type: AVM :a: :v: :m:,Type: Bug :bug:' --json 'title,url,body,comments' --repo $Repo | ConvertFrom-Json -Depth 100
+  $runs = gh run list --json 'url,workflowName,headBranch,startedAt' --limit $LimitNumberOfRuns --repo $Repo | ConvertFrom-Json -Depth 100
+  $workflowRuns = @{}
+  $issuesCreated = 0
+  $issuesCommented = 0
+  $issuesClosed = 0
+
+  foreach ($run in $runs) {
+    $runStartTime = [Datetime]::ParseExact($run.startedAt, 'MM/dd/yyyy HH:mm:ss', $null)
+
+    if (($run.headBranch -eq 'main') -and ($runStartTime -gt (Get-Date).AddDays(-$LimitInDays)) -and ($IgnoreWorkflows -notcontains $run.workflowName)) {
+      $runId = $run.url.Split('/') | Select-Object -Last 1
+      $runDetails = gh run view $runId --json 'conclusion,number' --repo $Repo | ConvertFrom-Json -Depth 100
+
+      if ($run.workflowName.StartsWith("avm.")) {
```

Comment:
I'd suggest to limit to module workflows. Not sure we want to open issues for platform ones

Reply:
Also platform pipelines should be green. Not sure if we should exclude them.

Reply:
We had challenges with that in carml and many stayed in a failing state, but let's see what others think as well. @ChrisSidebotham @AlexanderSehr

Reply:
I'd encourage inluding the platform pipelines. @eriqua could you elaborate on the problems you've experienced? Can't recall right now. 

On that note - at least right now they're all green ;) 
![image](https://github.com/Azure/bicep-registry-modules/assets/5365358/2f0899ab-b4bc-4381-8da0-09cc717a91c4)

Reply:
I remember we had challenges at least with Linter, opening issues although green state or something on that line. For the ones we have in BRM it should be fine. Question though @rahalan , if an avm issue is created for a failing platform pipeline, what would be the behavior of the assign issue pipeline?

Reply:
@eriqua nothing should happen, as the other pipeline looks for "[AVM Module Issue]" in the title